modules/apps/portal-security-crypto/portal-security-crypto-algorithm-pbkdf2/src/main/java/com/liferay/portal/security/crypto/algorithm/pbkdf2/factory/PBKDF2HashGeneratorFactory.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.security.crypto.algorithm.pbkdf2.factory;
+
+import com.liferay.portal.security.crypto.algorithm.pbkdf2.PBKDF2HashGenerator;
+import com.liferay.portal.security.crypto.generator.spi.HashGenerator;
+import com.liferay.portal.security.crypto.generator.spi.factory.HashGeneratorFactory;
+
+import org.json.JSONObject;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author arthurchan35
+ */
+@Component(
+	immediate = true, property = "crypto.hash.generator.factory=PBKDF2",
+	service = HashGeneratorFactory.class
+)
+public class PBKDF2HashGeneratorFactory implements HashGeneratorFactory {
+
+	@Override
+	public HashGenerator getGenerator(
+			String algorithmName, JSONObject algorithmMeta)
```

Comment:
hey @arthurchan35,

it looks like `algorithName` is not needed for creating the generator, only for getting the proper factory, so the signature for the factory can be just the `algorithmMeta` in all the accepted forms.

Reply:
Yes, for this particular PBKDF2HashGenerator, the algorithmName isnt necessary, but for HashGenerator like MessageDigestHashGenerator, an algorithmName is needed. Are you suggesting to hide algorithmName in side algorithmMeta?